core/tag.h

```diff
@@ -37,6 +37,27 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "platform.h"
 #include "limits.h"
 
+#ifdef GRAN_MICRO
+/* Conversion of time to nanoseconds. */
+#define NSEC(t) (t / 1000L)
```

Comment:
What happens if something gets scheduled 1 nanosecond in the future? I want to make sure that it ends up strictly in the future when it goes into the event queue (and not rounded down to the present).

Reply:
I believe we set up the workflow such that there is an error when instituting a delay under 3 microseconds due to precision reasons, so we shouldn't run into this problem.

Reply:
The error check for a delay under 3 microseconds is in nanosleep(), I think, which does not get involved when schedule() is called. I think what will happen is that the event will be scheduled one microstep in the future.